tests/wpunit/PostObjectQueriesTest.php

```diff
@@ -1659,4 +1659,53 @@ public function dataProviderRestrictedPosts() {
 
 	}
 
+	/**
+	 * Test the scenario where a post is assigned to an author
+	 * who is not a user on the site. This could happen for instance,
+	 * if the user was deleted, but their posts were never trashed
+	 * or assigned to another user.
+	 */
+	public function testQueryPostsWithOrphanedAuthorDoesntThrowErrors() {
+		global $wpdb;
+
+		$highest_user_id     = (int) $wpdb->get_var( "SELECT ID FROM {$wpdb->users} ORDER BY ID DESC limit 0,1" );
```

Comment:
Was this because of the inability to use `wp_delete_user()` ? I'd rather just include the core file that contains the function than do this.

Reply:
@CodeProKid I wan't able to use `wp_delete_user()`, since it takes all the posts that were assigned to the  user being deleted and trashes them. So by using it, we don't end up with the "orphaned" post scenario we're trying to test.

With that in mind, I revamped my code to create a new user, insert a post with then as the author, then delete them from the DB without trashing their posts, like this:
https://gist.github.com/kellenmace/612411db9e1fa7f5d98d1d4862b5fe2f

Ultimately though, that seemed like a lot of code just to end up with a post assigned to a bogus user ID, so I switched to my current implementation of just getting the highest user ID in the `wp_users` table and incrementing it by 1 to guarantee that the post being inserted is being assigned to a non-existent user ID.

Which approach to you prefer? At the end of the day, both of them result it a post being inserted with a bogus user ID as the author, so I think either suits the purpose of the test.

Reply:
@kellenmace thanks for slogging through this ridiculous issue with `wp_delete_user()`, you're a rockstar.

I think the current implementation with finding the highest user ID and adding 1 to it is a bit overkill for this test. Let's just go back to basics and use `999999` for the user. The DB is rolled back periodically throughout the tests (between individual unit test methods, as well as classes), so the chances of the ID overlapping are so insanely slim.

If someone else has a problem with that they can take it up with me! 😆

Reply:
@CodeProKid Cool! Sounds good. `999999` it is! I'll update it to that in a little bit.